bindings/python3/econf.py

```diff
@@ -0,0 +1,969 @@
+"""
+Econf provides functionality for interacting with Key-Value config files, like getting and setting values for read config files.
+
+For more information please have a look at the API
+"""
+import ctypes.util
+from enum import Enum
+from dataclasses import dataclass
+from typing import *
+from ctypes import *
+from collections import defaultdict
+
+LIBNAME = ctypes.util.find_library("econf")
+LIBECONF = CDLL(LIBNAME)
+
+
+@dataclass
+class EconfFile:
+    """
+    Class which points to the Key Value storage object
+    """
+
+    _ptr: c_void_p
+
+    def __del__(self):
+        free_file(self)
+
+
+class EconfError(Enum):
+    SUCCESS = 0
+    ERROR = 1
+    NOMEM = 2
+    NOFILE = 3
+    NOGROUP = 4
+    NOKEY = 5
+    EMPTYKEY = 6
+    WRITEERROR = 7
+    PARSE_ERROR = 8
+    MISSING_BRACKET = 9
+    MISSING_DELIMITER = 10
+    EMPTY_SECTION_NAME = 11
+    TEXT_AFTER_SECTION = 12
+    FILE_LIST_IS_NULL = 13
+    WRONG_BOOLEAN_VALUE = 14
+    KEY_HAS_NULL_VALUE = 15
+    WRONG_OWNER = 16
+    WRONG_GROUP = 17
+    WRONG_FILE_PERMISSION = 18
+    WRONG_DIR_PERMISSION = 19
+    ERROR_FILE_IS_SYM_LINK = 20
+    PARSING_CALLBACK_FAILED = 21
+
+
+_exceptions = defaultdict(Exception, {
```

Comment:
That now is a global variable, so the name should be something like in the first review: ECONF_EXCEPTION

The defaultdict is a good idea, but will hide the error in the code in case the errno does not match the enum. Most of the times this kind of hiding is a problem: become hard to debug and reason about the code.

Reply:
Actually the Enum throws a KeyError before that so it isn't hidden

Reply:
Right, so it is safe to move it to a dict to be more explicit

Reply:
Anyways I removed the defaultdict